internal/agenthttp/client.go

```diff
@@ -26,6 +34,29 @@ func NewClient(opts ...ClientOption) *http.Client {
 		opt(&conf)
 	}
 
+	// http client profile is used to switch between different http client implementations
+	// - stdlib: uses the standard library http client
+	switch conf.HTTPClientProfile {
+	case ClientProfileStdlib:
+		// Base any modifications on the default transport.
+		transport := http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
As far as I can tell https://go-review.googlesource.com/c/net/+/617655 is merged but not released. So using the default option would re-introduce the bug we fixed in #3005?

Reply:
@patrobinson Given 1.24 is coming out soon, with 1.24rc3 coming out a few days ago, I think I am going to wait rather than add this workaround.

I have confirmed this fix is bundled with 1.24 in https://github.com/golang/go/blob/release-branch.go1.24/src/net/http/h2_bundle.go#L7834-L7840 🤞🏻.